company/strategy.md

```diff
@@ -1,36 +1,129 @@
+- When you make a change to an API, you can include "metacode" that knows how to update all usages of the changed API. Sourcegraph will show you a compatibility report and propose those changes on all downstream callers along with your change.
+- You can discover and more easily use libraries, with statistics about usage patterns and a code marketplace.
+
+> Unlike the [1-year vision](#1-year-vision), our 3-year vision involves Sourcegraph modifying existing workflows and being proactive.
+
+### 10-year vision {#10-year-vision}
+
+> Problem statement: Software development is a black box to everyone else in an organization. Communication and collaboration in both directions is poor.
+
+Everyone involved in the software value chain (developers, sales, product, marketing, support, customers/users, etc.) uses Sourcegraph to coordinate work and share information.
+
+From a developer's point of view, they can go from the code for a particular feature to a list of all customers using it, all feedback and support requests about it, and all upcoming features that relate to it.
+
+From a customer's and salesperson's point of view, they can see all in-progress or planned features for the particular customer. For any given feature, they can see the current state of development.
+
+> Unlike the [3-year vision](#3-year-vision), our 10-year vision involves more kinds of people using Sourcegraph (not just people who write code directly).
+
+## Principles
+
+- Sourcegraph is universal code search, not universal "everything" search. Any additional data types in our search need to be relevant to the software development workflow.
+- We're a mass-market company and must maintain broad appeal. We want every developer, not just a specific niche audience, to use Sourcegraph.
+  - Any given developer will only pick one code search tool to use. Any given company will standardize on a single code search tool. Therefore, to avoid fragmentation, Sourcegraph should be not only *much* better than the alternatives, but also *not worse* in any significant way.
+  - We won't adopt polarizing public stances that would divide our audience.
+- Our product should strive to be fundamentally privacy-respecting and secure. This means that users don't need to trust us to verify that their data is private and secure.
+- Build on-ramps in our product to turn more people into frequent users, instead of building the product *for* infrequent users (which is a self-fulfilling prophecy).
+- We eventually want to be a platform that ties together all of the tools developers use.
+  - Other developer tools are partners, not as competitors.
+  - This entails designing more extensibility into our product (and documenting it more thoroughly) than what's needed today, in order to make it easy for other people to build on.
+
```

Comment:
Additional principle:
- Sourcegraph provides greater value the larger a software team. We are building not just for individual developers, but for development teams. We believe software development is increasingly a multiplayer game.